procbuild/builder.py

```diff
@@ -80,7 +80,172 @@ def checkout(repo, branch, build_path):
     return shell('git clone %s --branch %s --single-branch %s' %
                  (repo, branch, build_path), retry=4)
 
+class BuildError(Exception):
+    
+    def __init__(self, message):
+        self.message = f"A build error occurred during the self.{message} step."
+
+class BuildManager(object):
+    
+    def __init__(self, 
+                 user, 
+                 branch, 
+                 target, 
+                 master_branch='master', 
+                 log=None):
+        self.user = user
+        self.build_output = ''
+        self.status = 'fail'
+        self.branch = branch
+        self.target = target
+        self.master_branch = master_branch
+        self.build_status = 'Build started...'
+        self.build_pdf_path = ''
+        self.master_repo_path = joinp(cache(), 'scipy_proceedings')
```

Comment:
Should this be an input parameter as well?

Reply:
(my thinking is that the BuildManager should not be dependent on knowledge about the cache)

Reply:
Agreed on all of these counts. I also have a PRManager PR that i'm working on that changes how the cache works, I should probably make a WIP PR for that too… I just think it'll be easier to get these refactors done one at a time & this one was more isolated.

Reply:
NB: that comment was relevant because the cache should move into the PRManager since that's what's doing most of the cache management.